enterprise/api/v1/views.py

```diff
@@ -554,6 +555,146 @@ class EnrollmentModificationException(Exception):
+        """
+        Fetch the correct serializer class based on the subsidy type.
+        """
+        fulfillment_source_uuid = self.kwargs.get('fulfillment_source_uuid')
+
+        learner_credit_enrollments = models.LearnerCreditEnterpriseCourseEnrollment.objects.filter(
+            uuid=fulfillment_source_uuid
+        )
+        if len(learner_credit_enrollments):
+            return serializers.LearnerCreditEnterpriseCourseenrollmentReadOnlySerializer
+        licensed_enrollments = models.LicensedEnterpriseCourseEnrollment.objects.filter(
+            uuid=fulfillment_source_uuid
+        )
+        if len(licensed_enrollments):
+            return serializers.LicensedEnterpriseCourseEnrollmentReadOnlySerializer
+
+        raise ValidationError('No enrollment found for the given fulfillment source uuid.', code=HTTP_404_NOT_FOUND)
+
+    @permission_required(
+        'enterprise.can_access_admin_dashboard',
+        fn=lambda request, fulfillment_source_uuid: get_enterprise_customer_from_user_id(request.user.id)
+    )
+    def retrieve(self, request, fulfillment_source_uuid, *args, **kwargs):
+        """
+        Retrieve a single subsidized enrollment.
+            /enterprise/api/v1/subsidy-fulfillment/{fulfillment_source_uuid}/
+        """
+        try:
+            queryset = self.get_queryset()
+            fulfillment = get_object_or_404(queryset, uuid=fulfillment_source_uuid)
+            serializer_class = self.get_serializer_class()
+            serialized_object = serializer_class(fulfillment)
+        except ValidationError as exc:
+            return Response(
+                status=HTTP_404_NOT_FOUND,
+                data={'detail': exc.detail}
+            )
+        return Response(serialized_object.data)
+
+    @action(methods=['post'], detail=True)
+    @permission_required(
+        'enterprise.can_enroll_learners',
+        fn=lambda request, fulfillment_source_uuid: get_enterprise_customer_from_user_id(request.user.id)
+    )
+    def cancel_enrollment(self, request, fulfillment_source_uuid):
+        """
+        Cancel a single subsidized enrollment. Assumes fulfillment source has a valid enterprise enrollment.
+            /enterprise/api/v1/subsidy-fulfillment/{fulfillment_source_uuid}/cancel-enrollment/
+        """
+        try:
+            enrollment = get_object_or_404(
+                self.get_queryset(), uuid=fulfillment_source_uuid
+            )
+            if enrollment.is_revoked:
+                return Response(
+                    status=HTTP_400_BAD_REQUEST,
+                    data={'detail': 'Enrollment is already canceled.'}
+                )
+        except ValidationError as exc:
+            return Response(
+                status=HTTP_404_NOT_FOUND,
+                data={'detail': exc.detail}
+            )
+
+        try:
+            username = enrollment.enterprise_course_enrollment.enterprise_customer_user.username
+            enrollment_api.update_enrollment(
```

Comment:
I had the option of directly modifying the enrollment object via referenced property on the enterprise enrollment object. Since there existed a dedicated api + another endpoint that already leveraged the api instead of modifying the record, I opted to follow existing patterns